src/java.base/macosx/native/libosxsecurity/KeystoreImpl.m

```diff
@@ -511,95 +531,95 @@ static void addCertificatesToKeystore(JNIEnv *env, jobject keyStore)
+        paramBlock.accessRef = NULL;
+        paramBlock.keyUsage = CSSM_KEYUSE_ANY;
+        paramBlock.keyAttributes = CSSM_KEYATTR_RETURN_DEFAULT;
+
+        err = SecKeychainItemImport(cfDataToImport, NULL, &dataFormat, NULL,
+                                    0, &paramBlock, defaultKeychain, &createdItems);
+        if (cfDataToImport != NULL) {
+            CFRelease(cfDataToImport);
+        }
 
-    err = SecKeychainItemImport(cfDataToImport, NULL, &dataFormat, NULL,
-                                0, &paramBlock, defaultKeychain, &createdItems);
-    if (cfDataToImport != NULL) {
-        CFRelease(cfDataToImport);
-    }
+        if (err == noErr) {
+            SecKeychainItemRef anItem = (SecKeychainItemRef)CFArrayGetValueAtIndex(createdItems, 0);
 
-    if (err == noErr) {
-        SecKeychainItemRef anItem = (SecKeychainItemRef)CFArrayGetValueAtIndex(createdItems, 0);
+            // Don't bother labeling keys. They become part of an identity, and are not an accessible part of the keychain.
+            if (CFGetTypeID(anItem) == SecCertificateGetTypeID()) {
+                setLabelForItem(JavaStringToNSString(env, alias), anItem);
+            }
 
-        // Don't bother labeling keys. They become part of an identity, and are not an accessible part of the keychain.
-        if (CFGetTypeID(anItem) == SecCertificateGetTypeID()) {
-            setLabelForItem(JNFJavaToNSString(env, alias), anItem);
+            // Retain the item, since it will be released once when the array holding it gets released.
+            CFRetain(anItem);
+            returnValue = ptr_to_jlong(anItem);
+        } else {
+            cssmPerror("_addItemToKeychain: SecKeychainItemImport", err);
         }
 
-        // Retain the item, since it will be released once when the array holding it gets released.
-        CFRetain(anItem);
-        returnValue = ptr_to_jlong(anItem);
-    } else {
-        cssmPerror("_addItemToKeychain: SecKeychainItemImport", err);
-    }
-
-    if (createdItems != NULL) {
-        CFRelease(createdItems);
-    }
+        if (createdItems != NULL) {
+            CFRelease(createdItems);
+        }
 
-errOut:
-    if (rawData) {
-        (*env)->ReleaseByteArrayElements(env, rawDataObj, rawData, JNI_ABORT);
-    }
+    errOut:
+        if (rawData) {
+            (*env)->ReleaseByteArrayElements(env, rawDataObj, rawData, JNI_ABORT);
+        }
 
-    if (passwordStrRef) CFRelease(passwordStrRef);
-    if (passwordChars) {
-        // clear the password and release
-        memset(passwordChars, 0, passwordLen);
-        (*env)->ReleaseCharArrayElements(env, passwordObj, passwordChars,
-            JNI_ABORT);
+        if (passwordStrRef) CFRelease(passwordStrRef);
+        if (passwordChars) {
+            // clear the password and release
+            memset(passwordChars, 0, passwordLen);
+            (*env)->ReleaseCharArrayElements(env, passwordObj, passwordChars,
+                JNI_ABORT);
+        }
```

Comment:
Although you have it in the later code, here you are missing
 @catch (NSException *e) { 
     NSLog(@"%@", [e callStackSymbols]); 
 }

Reply:
Will add.

BTW, will these be shown on stdout or stderr? If so, is this a good idea?

Reply:
should be stderr. Whether to log is up to you. You could wrap it in something that checks for a debug build.
The idea is that if this ever happens there is a serious problem. NSException is only thrown (by the OS frameworks) in supposedly fatal scenarios.